src/civetweb.c

```diff
@@ -8107,6 +8112,29 @@ mg_websocket_write(struct mg_connection *conn,
 		headerLen = 10;
 	}
 
+	if(mask) {
+		/* add mask */
+		srand(time(NULL));
+		masking_key[0] = rand() & 0xff;
+		masking_key[1] = rand() & 0xff;
+		masking_key[2] = rand() & 0xff;
+		masking_key[3] = rand() & 0xff;
+
+		header[1] |= 0x80;
+		header[headerLen] = masking_key[0];
+		header[headerLen + 1] = masking_key[1];
+		header[headerLen + 2] = masking_key[2];
+		header[headerLen + 3] = masking_key[3];
+		headerLen += 4;
+		for (i = 0; i != dataLen; ++i) {
+			masked_data[i] = *data++ ^ masking_key[i&0x3];
+		}
```

Comment:
TODO: Do this for 32 bit elements instead of byte-wise (performance) - can be done by me later